CMakeLists.txt

```diff
@@ -177,8 +177,14 @@ install(
 # Process subdirectories
 add_subdirectory(doc)
 add_subdirectory(src)
-add_subdirectory(tests)
-add_subdirectory(samples)
+if (NOT (DEFINED XERCES_BUILD_FUZZERS))
```

Comment:
Why are these specifically excluded?  It might be nicer to only use the fuzzer-specific conditions on the fuzzer code itself.

Reply:
Fuzzing requires some compile-time instrumentation of the xerces-library code and certain libraries that need to be linked against the instrumented code later on. Since this instrumentation is not needed for the other tests, because I thought this way it would be cleaner. Do you still want to me to change?

Reply:
I am afraid I misunderstand your question though: Do you want just the if condition to change?

Reply:
I would prefer that the fuzzer keep its conditionals self-contained if possible.  That is, rather than disabling other unrelated features, it should only enable or disable the fuzzing functionality.  If you want to disable building of the tests and/or samples, I think they should have separate options to allow them to be enabled or disabled independently.

You could then do `-Dtests=OFF -Dsamples-OFF -Dfuzzer=ON` to achieve the same effect.

Also, the extra compile-time stuff using ExternalProject looks separate from building the fuzzer itself.  Maybe always build the fuzzer but have that additional stuff gated on the `fuzzer` option.  Or gate all of it.  But I would put that conditional stuff *inside* the fuzzer CMakeLists.txt rather than having any conditionals around include statements.